app/src/main/java/org/mozilla/fenix/components/history/PagedHistoryProvider.kt

```diff
@@ -90,6 +91,31 @@ class DefaultPagedHistoryProvider(
         }
     }
 
+    /**
+     * Returns the [History.Regular] corresponding to the given [History.Metadata] item.
+     *
+     * @param historyMetadata The [History.Metadata] to match.
+     * @return the [History.Regular] corresponding to the given [History.Metadata] item or null.
+     */
+    suspend fun getMatchingHistory(historyMetadata: History.Metadata): VisitInfo? {
+        val history = historyStorage.getDetailedVisits(
+            start = historyMetadata.visitedAt - BUFFER_TIME,
+            end = historyMetadata.visitedAt,
+            excludeTypes = listOf(
+                VisitType.NOT_A_VISIT,
+                VisitType.DOWNLOAD,
+                VisitType.REDIRECT_TEMPORARY,
+                VisitType.RELOAD,
+                VisitType.EMBED,
+                VisitType.FRAMED_LINK,
+                VisitType.REDIRECT_PERMANENT
+            )
+        )
+        return history.lastOrNull { it.url == historyMetadata.url }.also {
+            historyGroups = null
```

Comment:
So, that's technically in the wrong place. it should be right after you call `delete` for the group itself.

I think in practice it'll work out but it's pretty weird having this reset here in the lookup. I.e. this simple lookup function doesn't know anything about group removals, why should it reset the state.